### PR TITLE
multi: update log dep for sub-modules

### PR DESCRIPTION
**Pull request by ellemouton**
Update btclog dep to v2 for sqldb, kvdb, healthcheck and tor modules. 

Also move more logic into `NewDefaultLogHandlers` so that other repos can re-use this method to apply the logger build tags. 

EDIT: I've also added a commit to revert https://github.com/lightningnetwork/lnd/pull/8644 as that updated lead to some CI failures in the PR that points to this PR: https://github.com/lightningnetwork/lnd/actions/runs/11537815002/job/32115644260?pr=9213

### CODE REVIEW

**Comment by ellemouton** [comment]
Demonstration of CI passing for this [here](https://github.com/lightningnetwork/lnd/pull/9213)

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

File: healthcheck/go.mod [hunk 1]
```diff
@@ -1,9 +1,11 @@
 module github.com/lightningnetwork/lnd/healthcheck
 
-go 1.19
+go 1.21
```
[the remark this answers is not in the document]
Comment (ellemouton) [reply healthcheck/go.mod]: good idea - Ok i've unified the location of the version (to the bottom of the files) and removed any new toolchain additions (as I assume it's ideal to not have a `toolchain` version as that gets quite limiting?)

**Review by guggero — APPROVED** [review]
LGTM :tada:

**Review by bhandras — APPROVED** [review]

**Pull request merged**